modules/rostests/apitests/shell32/ShellState.cpp

```diff
@@ -0,0 +1,221 @@
+}
+
+START_TEST(ShellState)
+{
+    OSVERSIONINFO osinfo;
+    REGSHELLSTATE rss;
+    SHELLSTATE *pss;
+    SHELLFLAGSTATE FlagState;
+    LPBYTE pb;
+    int ret;
+
+    printf("GetVersion(): 0x%08lX", GetVersion());
+    printf("\n");
+
+    osinfo.dwOSVersionInfoSize = sizeof(osinfo);
+    GetVersionEx(&osinfo);
+    printf("osinfo.dwMajorVersion: 0x%08lX\n", osinfo.dwMajorVersion);
+    printf("osinfo.dwMinorVersion: 0x%08lX\n", osinfo.dwMinorVersion);
+    printf("osinfo.dwBuildNumber: 0x%08lX\n", osinfo.dwBuildNumber);
+    printf("osinfo.dwPlatformId: 0x%08lX\n", osinfo.dwPlatformId);
+    printf("\n");
+
+    printf("WINVER: 0x%04X\n", WINVER);
+    printf("_WIN32_WINNT: 0x%04X\n", _WIN32_WINNT);
+    printf("_WIN32_IE: 0x%04X\n", _WIN32_IE);
+    printf("NTDDI_VERSION: 0x%08X\n", NTDDI_VERSION);
+
+#ifdef UNICODE
+    printf("UNICODE\n");
+#else
+    printf("ANSI\n");
+#endif
+
+#ifdef _MSC_VER
+    printf("_MSC_VER: 0x%08X\n", int(_MSC_VER));
+#elif defined(__MINGW32__)
+    printf("__MINGW32__: 0x%08X\n", int(__MINGW32__));
+#elif defined(__clang__)
+    printf("__clang__: 0x%08X\n", int(__clang__));
+#else
+    #error Unknown compiler.
+#endif
+
+    printf("\n");
+
+    ok(sizeof(REGSHELLSTATE) >= 0x24, "sizeof(REGSHELLSTATE) was %d\n", (int)sizeof(REGSHELLSTATE));
+    printf("sizeof(SHELLSTATE): %d\n", (int)sizeof(SHELLSTATE));
+    printf("__alignof(SHELLSTATE): %d\n", (int)__alignof(SHELLSTATE));
+    printf("sizeof(SHELLFLAGSTATE): %d\n", (int)sizeof(SHELLFLAGSTATE));
+    printf("sizeof(CABINETSTATE): %d\n", (int)sizeof(CABINETSTATE));
+
+    pss = &rss.ss;
+    pb = (LPBYTE)pss;
+
+    ret = read_key(&rss);
+    if (ret)
+    {
+        return;
+    }
+
+    dump_pss(pss);
+    ok(rss.dwSize >= 0x24, "rss.dwSize was %ld (0x%lX).\n", rss.dwSize, rss.dwSize);
+
+#define DUMP_LONG(x) printf(#x ": 0x%08X\n", int(x));
+#define DUMP_BOOL(x) printf(#x ": %d\n", !!int(x));
+    DUMP_BOOL(pss->fShowAllObjects);
+    DUMP_BOOL(pss->fShowExtensions);
+    DUMP_BOOL(pss->fNoConfirmRecycle);
+    DUMP_BOOL(pss->fShowSysFiles);
+    DUMP_BOOL(pss->fShowCompColor);
+    DUMP_BOOL(pss->fDoubleClickInWebView);
+    DUMP_BOOL(pss->fDesktopHTML);
+    DUMP_BOOL(pss->fWin95Classic);
+    DUMP_BOOL(pss->fDontPrettyPath);
+    DUMP_BOOL(pss->fShowAttribCol);
+    DUMP_BOOL(pss->fMapNetDrvBtn);
+    DUMP_BOOL(pss->fShowInfoTip);
+    DUMP_BOOL(pss->fHideIcons);
+    DUMP_BOOL(pss->fWebView);
+    DUMP_BOOL(pss->fFilter);
+    DUMP_BOOL(pss->fShowSuperHidden);
+    DUMP_BOOL(pss->fNoNetCrawling);
+    DUMP_LONG(pss->lParamSort);
+    DUMP_LONG(pss->iSortDirection);
+    DUMP_LONG(pss->version);
+    DUMP_LONG(pss->lParamSort);
+    DUMP_LONG(pss->iSortDirection);
+    DUMP_LONG(pss->version);
+    DUMP_BOOL(pss->fSepProcess);
+#if 0
+    // FIXME
+    DUMP_BOOL(pss->fStartPanelOn);
+    DUMP_BOOL(pss->fShowStartPage);
+    DUMP_BOOL(pss->fIconsOnly);
+    DUMP_BOOL(pss->fShowTypeOverlay);
+    DUMP_BOOL(pss->fShowStatusBar);
+#endif
+
+    // FIXME
```

Comment:
What is this fixme for?

Reply:
It's for future use. Replaced with "#if NTDDI_VERSION >= 0x06000000".